cmd/upgrade/cluster/cmd.go

```diff
@@ -215,6 +225,59 @@ func run(cmd *cobra.Command, _ []string) {
 		interactive.Enable()
 	}
 
+	// if cluster is sts validate roles are compatible with upgrade version
+	_, isSTS := cluster.AWS().STS().GetRoleARN()
```

Comment:
Can you please add a validation like:
```
if !isSTS && mode != "" {
  reporter.Errorf("The 'mode' option is only supported for STS clusters")
  os.Exit(1)
}
```
But add it really early (before _any_ interactive prompt is shown)